### PR TITLE
Telephone-Number-Validator

### PR DESCRIPTION
**Pull request by latifa-wakili**
* on line(6) js code for passing freecodecomp errors I had to use (alert) but for esl\linters error I changed to this words(resultsDiv.innerText)

### CODE REVIEW

**Comment by S-BLH** [comment]
1. no linters error
2. used github workflow
3. great design implemented
4. clean code
5. your project is well designed and clear

**Comment by latifa-wakili** [comment]
@ahzamir

File: script.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    resultsDiv.innerText = `Valid US number: ${userInput}`;
+  } else {
+    resultsDiv.innerText = `Invalid US number: ${userInput}`;
+  }
+});
+
+document.getElementById('clear-btn').addEventListener('click', () => {
+  const resultsDiv = document.getElementById('results-div');
+  resultsDiv.innerText = '';
+});
```
Comment (ahzamir) [inline script.js:22]: - [OPTIONAL] - Try to add your Event Listeners in while DOM content is loading, it helps to add all of your event listeners to be ready to the action right when your page loads and prevents any side effects below is an example to help you do that:

```javascript
document.addEventListener('DOMContentLoaded', () => {
  //In here add all of your addEventListener functions
  // don't forget to add the constants that you need in your event listeners here as the scope of this one is higher than other codes
  
  // Here add all of you code that belongs to event listeners
})
```

File: script.js [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+  }
+
+  const regex = /^(1\s?)?(\(\d{3}\)|\d{3})[-\s]?\d{3}[-\s]?\d{4}$/;
+
+  if (regex.test(userInput)) {
+    resultsDiv.innerText = `Valid US number: ${userInput}`;
+  } else {
+    resultsDiv.innerText = `Invalid US number: ${userInput}`;
+  }
+});
```
Comment (ahzamir) [inline script.js:17]: - In here for empty input your code is not working instead you have to write your code like this:

```javascript
document.getElementById('check-btn').addEventListener('click', () => {
  const userInput = document.getElementById('user-input').value;
  const resultsDiv = document.getElementById('results-div');

  if (userInput === '') {
    resultsDiv.innerText = 'PleaSe provide a phone number!';
  } else {
    const regex = /^(1\s?)?(\(\d{3}\)|\d{3})[-\s]?\d{3}[-\s]?\d{4}$/;

    if (regex.test(userInput)) {
      resultsDiv.innerText = `Valid US number: ${userInput}`;
    } else {
      resultsDiv.innerText = `Invalid US number: ${userInput}`;
    }
  }
});
```
- In here I update the innerHtml to use = sign not ().
- And also I used else for your conditional statement.

**Review by ahzamir — APPROVED** [review]
### Hi @latifa-wakili 🙋‍♀️,

> Excellent work! 💯 You have done a great job implementing all the project requirements. 🥳 Your feature is complete! Congratulations!!!🎊

Check the [OPTIONAL] changes below implement them and then merge your branch.

- **[REQUIRED]** Share your project on this [GROUP](https://www.facebook.com/groups/kick-startdev), add a simple Proud Moment text and a screenshot of your project then besides that add your project link at the end of the text as a GitHub Source code reference and post.

- Share your Project on your LinkedIn account and add a proud moment text with a screenshot.

- Well Done great design Implemented. 🎯🎉👍
image

# Status: Approved ✅
There is nothing to say other than to merge your branch. 🎉🎯✅👍😊🥳

WinkThumbsUpGIF

 ## Highlights 🌞

🎉 Great Design Implemented 
✔️ Created a professional Repo (You created a repository with your project name)
✔️ Correct use of Gitflow (You created the development branch and made it your default branch, then created and committed your work on your feature branch)
✔️ Commit messages are meaningful
✔️ Setup the linters correctly (You added the correct linters config into your project according to the language you used on your code)
✔️ No linter errors (All linters checks have been passed 🟢)
✔️ You have followed the Telephone Number Validator project task and committed using the command line
✔️ Created the pull request (From your feature branch to the development branch you created a pull request and submitted the pull request link for review)
✔️ Good PR description (You added a meaningful description to your PR about your work and the features you added on this pull request)
✔️Your code is clean
✔️Your repo has a professional README.md file
✔️ Your partner has reviewed your repo
✔️Your design is responsive

**Cheers 🥂 and Happy coding!!! 👯 Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me @ahzamir in your question so I can receive the notification.**

**Pull request merged**